core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServices.java

```diff
@@ -440,9 +440,16 @@ private static AnnotatedHttpService build(String pathPrefix, Object object, Meth
         final Set<String> parameterNames = function.pathParamNames();
         final Set<String> expectedParamNames = pathMapping.paramNames();
         if (!expectedParamNames.containsAll(parameterNames)) {
-            Set<String> missing = Sets.difference(parameterNames, expectedParamNames);
+            final Set<String> missing = Sets.difference(parameterNames, expectedParamNames);
             throw new IllegalArgumentException("Missing @Param exists: " + missing);
         }
+        if (!parameterNames.containsAll(expectedParamNames)) {
+            final Set<String> missing = Sets.difference(expectedParamNames, parameterNames);
+            logger.warn("Some path variables of the method '" + method.getName() +
+                        "' of the class '" + clazz.getName() +
+                        "' do not have their corresponding @Param parameter. " +
+                        "They would not be automatically injected:" + missing);
```

Comment:
- `do not have their corresponding @Param parameter.` -> `do not have their corresponding parameters annotated with @Param.`?
- `injected:"` -> `injected: "` (missing space)